modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -136,19 +142,23 @@ func (k Keeper) GetAppVersion(ctx context.Context, portID, channelID string) (st
 }
 
 // GetActiveChannelID retrieves the active channelID from the store, keyed by the provided connectionID and portID
-func (k Keeper) GetActiveChannelID(ctx sdk.Context, connectionID, portID string) (string, bool) {
-	store := ctx.KVStore(k.storeKey)
+func (k Keeper) GetActiveChannelID(ctx context.Context, connectionID, portID string) (string, bool) {
+	store := k.storeService.OpenKVStore(ctx)
 	key := icatypes.KeyActiveChannel(portID, connectionID)
 
-	if !store.Has(key) {
+	bz, err := store.Get(key)
+	if len(bz) == 0 {
 		return "", false
 	}
+	if err != nil {
+		panic(err)
+	}
```

Comment:
error handling and panic before `len(bz)` check?

Reply:
moved it as the logic was wrong above